### ISSUE TITLE
cloning the dataProvider

### ISSUE THREAD

**Issue by mg1075** [amcharts/animate#4]
The documentation for this plugin says to clone the dataProvider in a certain manner:

> // Creates a deep copy of the old dataProvider
> var newDataProvider = JSON.parse(JSON.stringify(chart.dataProvider));

However, if the dataProvider contains dates, the dates objects will be rendered useless with this method as described in some of the answers of this SO post. 

- https://stackoverflow.com/questions/122102/what-is-the-most-efficient-way-to-deep-clone-an-object-in-javascript

Among the answers in the SO post are alternatives which do take into account Date objects.

**Comment by martynasma** [comment]
You're right, `JSON.stringify` will not clone Date objects correctly because JSON format does not allow object instances.

The example in our docs is provided for simplicity sake. You're absolutely free to choose any other deep cloning method that suits your requirements.

Thanks for the link, BTW!
